### PR TITLE
Issue #8181: Remove cli options -C and -W

### LINKED ISSUE #8181: Remove cli options -C and -W
Discussion at https://github.com/checkstyle/checkstyle/pull/8035#discussion_r410905300

### PR DESCRIPTION
**Pull request by gaurabdg**
Resolves #8181 
Discussion at https://github.com/checkstyle/checkstyle/pull/8035#discussion_r414608281

@romani @rnveach Please review and merge this as the parent PR dependts on this and I cant push changes till then.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java [hunk 1]
```diff
@@ -734,8 +732,6 @@ public boolean isLoggable(LogRecord record) {
          *
          * @noinspection CanBeFinal
          */
-        @Option(names = {"-W", "--tree-walker-threads-number"}, description = "(experimental) The "
-                + "number of TreeWalker threads (must be greater than zero)")
         private int treeWalkerThreadsNumber = DEFAULT_THREAD_COUNT;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java:735]: please put one sentence in javadoc of this field that it is skipped from CLI intentionally.

**Comment by strkkk** [comment]
@gaurabdg please also change commit message, `minor:` prefix is not for such changes.

**Comment by gaurabdg** [comment]
> @gaurabdg please also change commit message, `minor:` prefix is not for such changes.

@strkkk Yes, I was going to amend in my next commit after resolving the review items.

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java (hunk 1 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: @romani Done

**Comment by gaurabdg** [comment]
@romani @rnveach Please approve the issue, CI is failing.

**Pull request merged**